CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
 
-#### Non-standard color spaces
-For future consideration: support could be added for color space defined using the [CSS @color-profile rule](https://www.w3.org/TR/css-color-4/#at-profile).
+### WebGL 2 API Changes
 
-#### Compositing the canvas element
-Canvas contents are composited in accordance with the canvas element's style (e.g. CSS compositing and blending rules). The necessary compositing operations must be performed in an intermediate colorspace, the compositing space, that is implementation specific. The compositing space must have sufficient precision and a sufficiently wide gamut to guarantee no undue loss of precision or gamut clipping in bringing the canvas's contents to the display. Implementations should not expose color spaces that are unreasonble for the display.
+IDL Additions:
+<pre>
+partial dictionary WebGLContextAttributes {
+  CanvasColorSpace colorSpace = "srgb";
+  CanvasColorType colorType = "uint8";
+  GLenum pixelFormat = GL_NONE;
+};
+</pre>
 
-#### Feature detection
-2D rendering contexts are to expose a new getContextAttributes() method, that works much like the method of the same name on WebGLRenderingContext. The method returns the "actual context attributes" which represents the settings that were successfully applied at context creation time. The settings attribute reflects the result of running the algorithm for coercing the settings argument for 2D contexts, as well as the result of any fallbacks that may have happened as a result of options not being supported by the UA.
+Values stored in the WebGL backbuffers  are in the canvas's color space, as specified by the colorSpace context attribute.
 
-Web apps may infer that a user agent that does not implement getContextAttributes() does not support the colorSpace and pixelFormat attributes.
+The pixelFormat specifies the WebGL format of the backbuffer. If a value of GL_NONE is specified, then a default pixelFormat is determined based on the colorType and alpha attributes. The following table lists the supported combinations of colorFormat, alpha, and pixelFormat. The default pixelFormat for a given colorType and alpha pair is indicated by (default). All combinations not listed in this table are invalid.
 
-Note: An alternative approach that was considered was to augment the probablySupportsContext() API by making it check the second argument.  That approach is difficult to consolidate with how dictionary arguments are meant to work, where unsupported entries are just ignored.
+| colorType | alpha | Implied pixelFormat |
+| - | - | - |
+| "uint8" | false | RGB8 (default) |
+| "uint8" | false | SRGB8 |
+| "uint8" | true | RGBA8 (default) |
+| "uint8" | true | SRGB8_ALPHA8 |
+| "float16" | true | RGBA16F |
```

Comment:
If the web developer creates a canvas with `pixelFormat` of `GL_NONE` (or doesn't include it in the first place) and subsequently queries the context attributes after creation, will the `pixelFormat` member be `GL_NONE` or the value chosen from the table?  

Might be best to keep things simple and force developers to specify a `pixelFormat` from the table.

Reply:
I'll remove the GL_NONE part. The reason I added GL_NONE as a default value is because I thought a default value was required, and there is no default value that works universally.